src/driver/mpas_subdriver.F

```diff
@@ -151,10 +154,13 @@ end subroutine xml_stream_get_attributes
 
       call mpas_allocate_domain(domain_ptr)
 
+      domainID = domainID + 1
+      domain_ptr % domainID = domainID
```

Comment:
As written, the logic for setting `domain_ptr % domainID` seems odd to me. At least in my opinion, it might make more sense to increment `domainID` after assigning it to `domain_ptr % domainID`. That way, the initial value of `domainID` could be 0, which is the first domainID to be assigned.

Reply:
Yes, that makes way more sense.  I missed that when updating for multiple internally initialized MPAS domains.